homeassistant/components/lunatone/config_flow.py

```diff
@@ -0,0 +1,221 @@
+"""Config flow for Lunatone."""
+
+from __future__ import annotations
+
+import asyncio
+from enum import StrEnum
+import logging
+from typing import Any, Final
+
+import aiohttp
+from lunatone_rest_api_client import Auth, DALIScan, Info
+from lunatone_rest_api_client.models import StartScanData
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    SOURCE_RECONFIGURE,
+    SOURCE_USER,
+    ConfigFlow,
+    ConfigFlowResult,
+)
+from homeassistant.const import CONF_URL
+from homeassistant.helpers import config_validation as cv, selector
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DALIDeviceScanMethod(StrEnum):
+    """DALI device scan methods."""
+
+    DO_NOTHING = "do_nothing"
+    CURRENT_DEVICE_LIST = "current_device_list"
+    SYSTEM_EXTENSION = "system_extension"
+    NEW_INSTALLATION = "new_installation"
```

Comment:
What are the differences between these scan methods? Could it be automatic somehow? Or omitted? What happens if it is always set to new_installation, for example?

We should explain it in the docs too.

Reply:
I assume that having a constant periodic background scan would not be good, right? What about an integration service? Then the users can just call the service with the desired scan method when needed.